### ISSUE TITLE
cpp error

### ISSUE THREAD

**Issue by juzi5201314** [otiai10/gosseract#130]
ubuntu@VM-119-115-ubuntu:~$ go get -t github.com/otiai10/gosseract
# github.com/otiai10/gosseract
src/github.com/otiai10/gosseract/tessbridge.cpp: In function ‘void ClearPersistentCache(TessBaseAPI)’:
src/github.com/otiai10/gosseract/tessbridge.cpp:29:8: error: ‘class tesseract::TessBaseAPI’ has no member named ‘ClearPersistentCache’
   api->ClearPersistentCache();
        ^
src/github.com/otiai10/gosseract/tessbridge.cpp: In function ‘void SetPixImage(TessBaseAPI, PixImage)’:
src/github.com/otiai10/gosseract/tessbridge.cpp:57:12: error: ‘class tesseract::TessBaseAPI’ has no member named ‘GetSourceYResolution’
   if (api->GetSourceYResolution() < 70) {
            ^
ubuntu@VM-119-115-ubuntu:~$ 
...
```
ubuntu x64
```

```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/ubuntu"
GORACE=""
GOROOT="/usr/lib/go-1.6"
GOTOOLDIR="/usr/lib/go-1.6/pkg/tool/linux_amd64"
GO15VENDOREXPERIMENT="1"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0"
CXX="g++"
CGO_ENABLED="1"
```

```
go 1.6.2
```

```
tesseract 3.02.02
 leptonica-1.72
```

**Comment by otiai10** [comment]
# Reproducible Docker

```dockerfile
FROM ubuntu:16.04

ARG TESS="3.02.02"
ARG LEPTO="v1.72"
ARG GO="1.6.2"

RUN apt-get update -qq
RUN apt-get install -yq \
  git \
  wget \
  make \
  autoconf \
  automake \
  libtool \
  autoconf-archive \
  pkg-config \
  libpng-dev \
  libjpeg-dev \
  libtiff-dev \
  zlib1g-dev \
  libicu-dev \
  libpango1.0-dev \
  libcairo2-dev

ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/lib

# Compile Leptonica
WORKDIR /
RUN mkdir -p /tmp/leptonica \
  && wget -nv https://github.com/DanBloomberg/leptonica/archive/${LEPTO}.tar.gz \
  && tar -xzf ${LEPTO}.tar.gz -C /tmp/leptonica \
  && mv /tmp/leptonica/* /leptonica
WORKDIR /leptonica
RUN chmod a+x ./autobuild ./configure
RUN autoreconf -i \
  && ./autobuild \
  && ./configure \
  && make --quiet \
  && make install

# Compile Tesseract
WORKDIR /
RUN mkdir -p /tmp/tesseract \
  && wget -nv https://github.com/tesseract-ocr/tesseract/archive/${TESS}.tar.gz \
  && tar -xzf ${TESS}.tar.gz -C /tmp/tesseract \
  && mv /tmp/tesseract/* /tesseract
WORKDIR /tesseract
RUN ./autogen.sh \
  && ./configure \
  && make --quiet \
  && make install

# Recover location
WORKDIR /

# Load languages
RUN wget -nv https://github.com/tesseract-ocr/tessdata/raw/master/eng.traineddata -P /usr/local/share/tessdata
RUN wget -nv https://github.com/tesseract-ocr/tessdata/raw/master/jpn.traineddata -P /usr/local/share/tessdata

# Install Go1.9.1
RUN wget -nv https://storage.googleapis.com/golang/go${GO}.linux-amd64.tar.gz \
  && tar -xzf go${GO}.linux-amd64.tar.gz
ENV GOROOT=/go

# Prepare GOPATH
RUN mkdir /gopath
ENV GOPATH=/gopath
ENV PATH=${PATH}:${GOROOT}/bin:${GOPATH}/bin

# Dependencies for tests
RUN go get github.com/otiai10/mint
RUN go get golang.org/x/net/html

# Mount source code of gosseract project
ADD . ${GOPATH}/src/github.com/otiai10/gosseract

ENTRYPOINT go test github.com/otiai10/gosseract
```

**Comment by otiai10** [comment]
https://github.com/tesseract-ocr/tesseract/commit/88ea81c89e2f71c428630cbe7b524e76be7a951d

**Comment by otiai10** [comment]
@juzi5201314 Thank you for reporting. I think we should no longer support tesseract 3.02.02, because `ClearPersistentCache` was [implemented on 3.03](https://github.com/tesseract-ocr/tesseract/commit/88ea81c89e2f71c428630cbe7b524e76be7a951d).

**Comment by juzi5201314** [comment]
ok,thanks
